client/js/components/data_page.jsx

```diff
@@ -1,14 +1,18 @@
 var React = require("react");
 var _ = require("lodash");
 var reactBootStrap = require("react-bootstrap");
+var AppStore = require("../store/app_store");
 
 var UploadButton = React.createClass({
-  fileHandler: function(event) {
-    console.log(this.files);
-  },
   componentDidMount: function() {
+    var schemaName = this.props.schemaName;
+    function fileHandler() {
+      var file = this.files.item(0);
```

Comment:
I think so. This function executes on a change event in the file upload. I think the event will only trigger if a file is uploaded.